File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/interactor/lock/disconnect/DisconnectAllLockUseCase.java
package com.lattis.ellipse.domain.interactor.lock.disconnect;

import com.lattis.ellipse.domain.executor.PostExecutionThread;
import com.lattis.ellipse.domain.executor.ThreadExecutor;
import com.lattis.ellipse.domain.interactor.lock.base.BaseLockUseCase;
import com.lattis.ellipse.domain.interactor.lock.connect.ConnectToLockUseCase;
import com.lattis.ellipse.domain.repository.BluetoothRepository;
import com.lattis.ellipse.domain.repository.LockRepository;

import javax.inject.Inject;

import io.reactivex.Observable;

/**
 * Created by ssd3 on 5/5/17.
 */

public class DisconnectAllLockUseCase extends BaseLockUseCase<Boolean> {

    private final String TAG = ConnectToLockUseCase.class.getName();

    @Inject
    DisconnectAllLockUseCase(ThreadExecutor threadExecutor,
                         PostExecutionThread postExecutionThread,
                         BluetoothRepository bluetoothRepository, LockRepository lockRepository) {
        super(threadExecutor, postExecutionThread,bluetoothRepository,lockRepository);
    }

    @Override
    protected Observable<Boolean> buildUseCaseObservable() {
        return disconnectAllLocks();
    }

}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/usecase/lock/disconnect/DisconnectAllLockUseCase.kt
package com.lattis.domain.usecase.lock.disconnect

import com.lattis.domain.executor.PostExecutionThread
import com.lattis.domain.executor.ThreadExecutor
import com.lattis.domain.repository.BluetoothRepository
import com.lattis.domain.repository.LockRepository
import com.lattis.domain.repository.SaSOrPSLockRepository
import com.lattis.domain.usecase.lock.base.BaseLockUseCase
import com.lattis.domain.usecase.lock.connect.ConnectToLockUseCase
import io.reactivex.rxjava3.core.Observable
import javax.inject.Inject

class DisconnectAllLockUseCase @Inject internal constructor(
    threadExecutor: ThreadExecutor,
    postExecutionThread: PostExecutionThread,
    bluetoothRepository: BluetoothRepository, lockRepository: LockRepository,
    saSOrPSLockRepository: SaSOrPSLockRepository
) : BaseLockUseCase<Boolean>(
    threadExecutor,
    postExecutionThread,
    bluetoothRepository,
    lockRepository,
    saSOrPSLockRepository
) {

    fun withLockVendor(vendor:LockVendor):DisconnectAllLockUseCase{
        this.lockVendor = vendor
        return this
    }

    override fun buildUseCaseObservable(): Observable<Boolean> {
        return disconnectAllLocks(lockVendor)
    }
}